modules/lib/lib-project/src/main/resources/lib/xp/project.js

```diff
@@ -0,0 +1,173 @@
+ * Deletes an existing Content Project. This will delete all of the data inside the project repository.
+ *
+ * @param {Object} params JSON with the parameters.
+ * @param {string} params.id Unique project id to identify the project.
+
+ *
+ * @returns {boolean} True if project was successfully deleted.
+ */
+exports.delete = function (params) {
+    var bean = __.newBean('com.enonic.xp.lib.project.DeleteProjectHandler');
+    bean.id = required(params, 'id');
+    return __.toNativeObject(bean.execute());
+};
+
+/**
+ * Returns an existing Content Project.
+ *
+ * @param {Object} params JSON with the parameters.
+ * @param {string} params.id Unique project id to identify the project.
+
+ *
+ * @returns {Object} Content Project.
+ */
+exports.get = function (params) {
+    var bean = __.newBean('com.enonic.xp.lib.project.GetProjectHandler');
+    bean.id = required(params, 'id');
+    return __.toNativeObject(bean.execute());
+};
+
+/**
+ * Returns all Content Projects.
+ *
+
+ *
+ * @returns {Object[]} Array of Content Projects.
+ */
+exports.list = function () {
+    var bean = __.newBean('com.enonic.xp.lib.project.ListProjectsHandler');
+    return __.toNativeObject(bean.execute());
+};
+
+/**
+ * Adds permissions to an existing Content Project.
+ *
+ * @param {Object} params JSON with the parameters.
+ * @param {string} params.id Unique project id to identify the project.
+ * @param {Object.<string, string[]>} params.permissions Project permissions to add. 1 to 5 properties where key is a role and value is an array of principals.
+ * @param {string} params.permissions.role - Role id (one of `owner`, `editor`, `author`, `contributor`, `viewer`)
+ * @param {Object[]} params.permissions.principals - Array of principals to add to this role
+
+ *
+ * @returns {Object} All current project permissions.
+ */
+exports.addPermissions = function (params) {
+    var bean = __.newBean('com.enonic.xp.lib.project.AddProjectPermissionsHandler');
+    bean.id = required(params, 'id');
+    bean.permissions = __.toScriptValue(params.permissions);
+    return __.toNativeObject(bean.execute());
+};
+
+/**
+ * Removes permissions from an existing Content Project.
+ *
+ * @param {Object} params JSON with the parameters.
+ * @param {string} params.id Unique project id to identify the project.
+ * @param {Object.<string, string[]>} params.permissions Project permissions to delete. 1 to 5 properties where key is a role and value is an array of principals.
+ * @param {string} params.permissions.role - Role id (one of `owner`, `editor`, `author`, `contributor`, `viewer`)
+ * @param {Object[]} params.permissions.principals - Array of principals to delete from this role
+
+ *
+ * @returns {Object} All current project permissions.
+ */
+exports.removePermissions = function (params) {
+    var bean = __.newBean('com.enonic.xp.lib.project.RemoveProjectPermissionsHandler');
+    bean.id = required(params, 'id');
+    bean.permissions = __.toScriptValue(params.permissions);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['__' is not defined.](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5511138)